vendor/github.com/grafana/dskit/cache/memcached_client.go

```diff
@@ -264,7 +259,7 @@ func newMemcachedClient(
 ) (*memcachedClient, error) {
 	addressProvider := dns.NewProvider(
 		logger,
-		prometheus.WrapRegistererWithPrefix("thanos_memcached_", reg),
+		prometheus.WrapRegistererWithPrefix("memcached_", reg),
```

Comment:
This is a breaking change for Mimir. It's fine doing this change in dskit, but then in Mimir you should wrap the registerer to add the `thanos_` prefix back.